Expand docs on App Transport Security

andrewsardone opened the pull request:
Building off of the work in this PR https://github.com/facebook/react-native/pull/5284, this…

- Adds a note int he docs about adding a `localhost` exception to an embedded app's `Info.plist`
- ~~Remove the total disabling of ATS from the react-native generated iOS project in favor of a `localhost` exception~~ see https://github.com/facebook/react-native/pull/5355 per this request: https://github.com/facebook/react-native/pull/5290#issuecomment-172213709

satya164 commented on docs/EmbeddedAppIOS.md:
````diff
@@ -161,6 +161,37 @@ In root directory, we need to start React Native development server.
+
+If you don't do this when connecting over http to your server you will see this error:
+```
+Could not connect to development server.
````
Prolly we shouldn't write the whole error since it's likely to change. You can write,

```
If you don't do this, you will see the error - `Could not connect to development server.` when connecting to your server over http.
```

andrewsardone replied:
Good idea, @satya164. Updated here: https://github.com/andrewsardone/react-native/commit/d23d48a710188265bf6f6a186536caa0911fbcf5

satya164 commented:
Awesome. Can you squash the commits to one before we merge?

andrewsardone commented:
@satya164: no problem! I squashed all docs-related commits into one, the original commit by @hfossli: https://github.com/andrewsardone/react-native/commit/3ca04fa204b5465f4fb42009ba6c48230fee46ae

But I left a second commit for tweaking the generated Xcode project's ATS settings: https://github.com/facebook/react-native/commit/0b5a575d2fde549997bb5615488ea6eefbe9e88a. Since this wasn't documentation related, but code/project related, I didn't think it should be squashed. If you disagree, just let me know and I'll put ‘em all into one commit. Thanks!

satya164 commented:
On that, I'm sorry for the trouble. But you'll need to split this PR into two, one for the docs, and one for the change in the template. This is because the process for merging them are different. :(

andrewsardone commented:
@satya164, not a problem. I have rebased this guy against `origin/master` and removed the [ATS settings for the generated project commit](https://github.com/facebook/react-native/commit/0b5a575d2fde549997bb5615488ea6eefbe9e88a). I have opened up a new PR for that change: https://github.com/facebook/react-native/pull/5355

satya164 commented:
Thank you.
